lightningd/peer_control.c

```diff
@@ -748,6 +748,31 @@ static void json_add_channel(struct lightningd *ld,
 			type_to_string(tmpctx, struct channel_id, &channel->cid));
 	json_add_txid(response, "funding_txid", &channel->funding_txid);
 
+	if (channel->state == DUALOPEND_AWAITING_LOCKIN) {
+		struct channel_inflight *inflight;
+		u32 last_feerate, next_feerate, feerate;
+		u8 feestep;
+
+		last_feerate = channel_last_funding_feerate(channel);
+		assert(last_feerate > 0);
+		next_feerate = last_feerate + last_feerate / 4;
+
+		inflight = list_top(&channel->inflights,
+				    struct channel_inflight, list);
+		feerate = inflight->funding->feerate;
+
+		for (feestep = 0; feerate < next_feerate; feestep++)
+			feerate += feerate / 4;
+
+		json_add_string(response, "last_feerate",
+				tal_fmt(tmpctx, "%d%s", last_feerate,
+					feerate_style_name(FEERATE_PER_KSIPA)));
+		json_add_string(response, "next_feerate",
+				tal_fmt(tmpctx, "%d%s", next_feerate,
+					feerate_style_name(FEERATE_PER_KSIPA)));
+		json_add_num(response, "fee_step", feestep);
```

Comment:
Hmm, this calculation confused me.  Perhaps rename "inflight" to "initial" and then  it's clearer that you're counting steps to get to the current one (using < next_feerate rather than == last_feerate is also confusing to me).